### PR TITLE
MM-15569 Fixes failing test on TestGetPostsForChannelAroundLastUnread

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
This PR intends to finalize unread API so the 'post-list-improvements' branch could finally submitted against master.
It includes:
- fix to failing TestGetPostsForChannelAroundLastUnread
- reverts unnecessary change to GetPostsSince - https://github.com/mattermost/mattermost-server/pull/11039/commits/15fb9a4238a568a05bb3a0a874c59169224fd2aa
- migrate the new functions - GetPostAfterTime and GetPostBeforeTime - to sync by default - https://github.com/mattermost/mattermost-server/pull/11039/commits/ce3540428fb137bafbf3a69949e16a53a28c297e

This PR was originally submitted as https://github.com/mattermost/mattermost-server/pull/10861.  However, I messed up that PR and `unread-posts-api` branch.

I copied over the comments made by HH.  Sorry for inconvenience in reviewing this PR.

#### Ticket Link
Jira ticket: [MM-15569](https://mattermost.atlassian.net/browse/MM-15569)

### CODE REVIEW

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func (s *SqlPostStore) InvalidateLastPostTimeCache(channelId string) {
 
 func (s *SqlPostStore) GetEtag(channelId string, allowFromCache bool) string {
 	if allowFromCache {
-		if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok && cacheItem.(int64) > 0 {
+		if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok {
```
Comment (saturninoabril) [inline store/sqlstore/post_store.go]: [HH comment](https://github.com/mattermost/mattermost-server/pull/10861#discussion_r286046656): What's the reason for this change? I don't understand why that extra check was there in the first place, but I'm not sure why it would we removed

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -555,8 +555,8 @@ func (s *SqlPostStore) GetPostsSince(channelId string, time int64, limit int, al
 			    WHERE
 			        UpdateAt > :Time
 			            AND ChannelId = :ChannelId
-			    LIMIT :Limit) temp_tab))
-			ORDER BY CreateAt DESC`,
+				LIMIT 1000) temp_tab))
+			ORDER BY CreateAt ASC LIMIT :Limit`,
```
Comment (saturninoabril) [inline store/sqlstore/post_store.go]: [HH comment](https://github.com/mattermost/mattermost-server/pull/10861#discussion_r286047410): Why change the order from descending to ascending?

Comment (saturninoabril) [reply store/sqlstore/post_store.go]: __My original thought is that:__ I changed it 'cause I think the old implementation of `GetPostsSince` does not sound accurate.  The existing limit of 1000 seems to mean return "all" posts from a certain time to new post.  We don't have any issue with it being "all" since our client usually don't ask for a very far "since" and as a result it returns less than 1000 posts.

However with unread API, it only gets limited number of posts around last unread (or a period that the channel was last viewed) and returns a number of post (say 60) from such period (old) to recent post).  If we keep DESC and put limit of 60 on it, it will return 60 posts starting from recent to old, and as a result it will going to have missing messages in between.

However, upon looking closer on the changes I've made, I realized that I should have not touch the GetPostSince and should just stick on GetPostsAfter.  I'll revise the PR...

File: store/sqlstore/post_store.go (hunk 1 above)
Comment (saturninoabril) [reply store/sqlstore/post_store.go]: I initially got a failing test and I thought it was related to it.  Reverted on last commit - https://github.com/mattermost/mattermost-server/pull/11039/commits/fe6bfe81f280a3d6746e1c99e9a0e7093217eea8.
The reason I added that before was that there's instance the cacheItem is equal to `0` and in turn, it returns outdated posts.  Honestly, I was not able to replicate the original issue since there's no client yet that's using this unread API.  That extra check wouldn't hurt but I'll keep an eye about the original issue.

Comment (hmhealey) [reply store/sqlstore/post_store.go]: Okay, this is fine then

File: store/sqlstore/post_store.go (hunk 2 above)
Comment (hmhealey) [reply store/sqlstore/post_store.go]: Are you still making that change to use `GetPostsAfter` for `GetPostsForChannelAroundLastUnread`?

Comment (saturninoabril) [reply store/sqlstore/post_store.go]: Yes, I did it here - https://github.com/mattermost/mattermost-server/pull/11039/files#diff-317b12c12f71e311f13813107f4d8550R807

Comment (hmhealey) [reply store/sqlstore/post_store.go]: Are these changes to `GetPostsSince` still needed then? I'm just leery about changing this function since it already works a bit unexpectedly

Comment (saturninoabril) [reply store/sqlstore/post_store.go]: No longer needed for unread API.  I've reverted all changes to GetPostSince in reference to master.
